sympy/core/function.py

```diff
@@ -706,6 +706,7 @@ def _eval_nseries(self, x, n, logx):
             if e == e1:
                 #for example when e = sin(x+1) or e = sin(cos(x))
                 #let's try the general algorithm
+                e=e.simplify()
```

Comment:
```suggestion
                if len(e.args) == 1:
                     # issue 14411
                     e = e.func(e.args[0].cancel())
```
It seems that this simplification will suffice to fix 14411 without affecting special functions.